### PR TITLE
Update ICS 20 readme

### PR DESCRIPTION
**Pull request by thpani**
Closes https://github.com/informalsystems/apalache-team-services/issues/13

Extend the ICS 20 README to reflect the current state of the spec.
This has become more of a tutorial-style text, but I think this fits well with our IBC outreach objective.

### CODE REVIEW

File: examples/cosmos/ics20/README.md [hunk 1]
```diff
@@ -1,13 +1,157 @@
...
+```json
+>>> chainStates.get("A").inAcknowledgements
+Set({ errorMessage: "", success: true })
+>>> receiveAck("A")
+true
+```
 
-- `bank` contains an abstract bank module that supports minting, burning, and
-  transfer of fungible tokens.
+To take the protocol through a complete transfer, in a single step, use the `sendTransfer` action:
```
Comment (bugarela) [inline examples/cosmos/ics20/README.md:121]: `sendTransfer` is actually a run
```suggestion
To take the protocol through a complete transfer, in a single step, use the `sendTransfer` run:
```

File: examples/cosmos/ics20/README.md [hunk 2]
```diff
@@ -39,5 +183,12 @@ if source {
 }
 ```
 
+### ICS 20
```
Comment (bugarela) [inline examples/cosmos/ics20/README.md:186]: Is this header meant to have this name? I found it confusing.

**Review by bugarela — APPROVED** [review]
Great format, very informative!

File: examples/cosmos/ics20/README.md [hunk 3]
```diff
@@ -1,13 +1,157 @@
...
+>>>
+```
+
+`ics20Test` sets up three chains `"A"`, `"B"` and `"C"`, all connected to each other via a channel `"channelToX"`.
+
+
+It also provides an action `init` that sets up the initial state of the protocol. This assigns 100 ATOM to account `"alice"` on chain `"A"`.
+The state of each chain is available as `chainStates.get(`chain`)`, and can be inspected interactively:
+
+```json
```
[the remark this answers is not in the document]
Comment (thpani) [reply examples/cosmos/ics20/README.md]: Oh dang, it looked so nice in VSCode, I didn't bother to check on GH... 🤪

File: examples/cosmos/ics20/README.md (hunk 2 above)
Comment (thpani) [reply examples/cosmos/ics20/README.md]: Could be confusing, I agree. I was trying to group by file, so I've just replaced it with the literal file name.

File: examples/cosmos/ics20/README.md [hunk 4]
```diff
@@ -1,20 +1,164 @@
...
+Instead of exploring the protocol interactively, we can encode longer protocol runs, together with assertions that assure correct behavior, as Quint tests. Use the Quint test runner to execute the tests in [`ics20.qnt`](./ics20.qnt):
+
+```sh
+$ quint test --main=ics20Test examples/cosmos/ics20/ics20.qnt
+
+  ics20Test
+    ok ABCACBATest passed 10000 test(s)
+    ok TransferTest passed 10000 test(s)
+    ok FailedAckTest passed 10000 test(s)
+    ok TimeoutTest passed 10000 test(s)
```
Comment (konnov) [inline examples/cosmos/ics20/README.md:151]: is there non-determinism inside? I wonder why the engine is running 10000 tests

File: examples/cosmos/ics20/README.md [hunk 5]
```diff
@@ -1,20 +1,164 @@
...
+Following the ICS 20 protocol, 1 ATOM has been moved out of Alice's account, and into a module-specific escrow account:
+
+```bluespec
+>>> chainStates.get("A")
+{
+  bank: Map(
+    "alice" -> Map({ baseDenom: "atom", path: [] } -> 99),
+    "escrow_account" -> Map({ baseDenom: "atom", path: [] } -> 1)
+  ),
+  [...]
```
Comment (konnov) [inline examples/cosmos/ics20/README.md:70]:
```suggestion
  // ...
```

File: examples/cosmos/ics20/README.md [hunk 6]
```diff
@@ -1,20 +1,164 @@
...
 
 ## Assumptions
 
 Where the [natural language spec][ICS 20] is imprecise or underspecified, we
 make the following assumptions:
 
-### Bank
+### `bank.qnt`
 
 In general, the bank module is underspecified. We provide an abstract,
```
Comment (konnov) [inline examples/cosmos/ics20/README.md:163]:
```suggestion
In general, the bank module is not specified in detail. We provide an abstract,
```

**Review by konnov — APPROVED** [review]
Looks great!

File: examples/cosmos/ics20/README.md (hunk 4 above)
Comment (thpani) [reply examples/cosmos/ics20/README.md:151]: what constitutes non-determinism?
there's `oneOf` for picking a packet, but in the test the set it is picking out of always has cardinality 1.

**Pull request merged**